llvm/lib/YkIR/YkIRWriter.cpp

```diff
@@ -1742,7 +1743,10 @@ class YkIRWriter {
     OutStreamer.emitSizeT(M.size());
     // funcs:
     for (llvm::Function &F : M) {
-      serialiseFunc(F);
+      // Skip cloned functions
```

Comment:
Is it enough to just skip the cloned functions? What about the rest of the module, e.g. `M.size()` above. Is this still serialised somewhere? Do we not need to skip this in the de-serialiser?

Reply:
I see that in a later commit you updated how `M.size()` is calculated. But now I'm confused. Do the cloned functions live in the same module as the original? Or do we have two separate modules?

Reply:
Maybe I got confused here, but I thought that we don't need the cloned functions in the IR?
https://github.com/ykjit/ykllvm/pull/202#issuecomment-2378723618

In the current version of this PR both function versions (original and cloned, excluding the functions that have reference taken to them) live in the same module, but we serialize only the non-cloned version in `YkIRWriter`.

Reply:
Ah I see. I think that answers my question. Though now I'm curious how we are going to optimise one version of the function while not optimising the other. That's something we need to think about soon.